File: Electro_Om_250_v2025.pas
Program Electro_Om_250_v2025;
Const 
    Freq = 250;
    R = 1;
Var 
    t, y : Real;
    HalfFreq : Real;
    Step : Integer;
    j : Smallint;
Begin;
    HalfFreq := Freq / 2;
    ReadLn (t); t := t + 24*60*60*2;
    if ((t = Trunc (t)) and ((t / HalfFreq) = 0))
       then begin
          y := 0;
       end
       else 
       begin
          Step := Trunc (t / HalfFreq);
          if ((Step / 2) = 0)
          then
              j :=1;
          else
              j := -1;
          end;
          y := j * sqrt ((R*R) - (t*t));
       end; 
    WriteLn (y);
end.

 
{ by Tovarisch Trunaev, 2025, Russia, Uray }

File: Electro_Om_250_v2025.pl
my $freq = 250;
my $r = 1;

my ($t, $y);
my $half_freq;
my $step;
my $j;

$half_freq = $freq / 2;
$t = <>;
$t += 24*60*60*2;

if (($t == int ($t)) && (($t / $half_freq) == 0)) {
	$y = 0;
} else { 
	$step = int ($t / $half_freq);
	if (($step / 2) == 0) {
		$j = 1;
	} else {
		$j = -1;
    }
	$y = $j * sqrt (($r*$r) - ($t*$t));
}
say ($y);

# by Tovarisch Trunaev, 2025, Russia, Uray
